packages/scene-composer/src/components/panels/AddOrRemoveOverlayButton.tsx

```diff
@@ -0,0 +1,74 @@
+import * as THREE from 'three';
+import React, { useCallback, useContext, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import { Box, Button } from '@awsui/components-react';
+
+import { COMPOSER_FEATURES, KnownComponentType } from '../../interfaces';
+import { IDataOverlayComponentInternal, useStore } from '../../store';
+import { sceneComposerIdContext } from '../../common/sceneComposerIdContext';
+import { findComponentByType } from '../../utils/nodeUtils';
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { Component } from '../../models/SceneModels';
+
+export const AddOrRemoveOverlayButton: React.FC = () => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const selectedSceneNodeRef = useStore(sceneComposerId)((state) => state.selectedSceneNodeRef);
+  const getSceneNodeByRef = useStore(sceneComposerId)((state) => state.getSceneNodeByRef);
+  const addComponentInternal = useStore(sceneComposerId)((state) => state.addComponentInternal);
+  const removeComponent = useStore(sceneComposerId)((state) => state.removeComponent);
+  const overlayEnabled = getGlobalSettings().featureConfig[COMPOSER_FEATURES.Overlay];
+  const selectedSceneNode = getSceneNodeByRef(selectedSceneNodeRef);
+  const intl = useIntl();
+
+  const isTagComponent = useMemo(
+    () => !!findComponentByType(selectedSceneNode, KnownComponentType.Tag),
+    [selectedSceneNode],
+  );
+
+  const isOverlayComponent = useMemo(
+    () => !!findComponentByType(selectedSceneNode, KnownComponentType.DataOverlay),
+    [selectedSceneNode],
+  );
+
+  const onClick = useCallback(() => {
+    if (!selectedSceneNodeRef) return;
+
+    if (isOverlayComponent) {
+      removeComponent(
+        selectedSceneNodeRef,
+        findComponentByType(selectedSceneNode, KnownComponentType.DataOverlay)!.ref,
+      );
+      return;
+    }
```

Comment:
Blocker: Add comments explaining why `useMemo` and `useCallback` are used.

Reply:
will remove useMemo since it's not needed. but I don't see a reason to add comment for explaining why useCallback is use

Reply:
This should not be a blocker, `useCallback` for instance prevents unnecessary redeclaration of functions during rerender. While it's true that removing unnecessary rerenders results in this being effectively a no-op, this is a mechanism that enforces it, and it's generally best practice to wrap all event handlers in `useCallback`.

`useMemo` is also correctly applied in this case, there's no reason to justify it's usage when leveraging standard library functions correctly. 

Also, the useMemo's in this case are actually utilized here, and I don't think they should be removed. `findComponentByType` is a recursive operation that can be expensive. The useMemo ensures this operation is only performed when the selecetdSceneNode changes, but there are several other reasons for this component to change that results in additional processing that should be avoided.

The feedback could be an indication that we should separate out some of those behaviors into smaller components, but at least as written, useMemo appears to be applied correctly.

Reply:
I disagree. It is not a large ask to add a comment for its usage. One person thinks it can be removed, the other doesn't. A single line comment would not have this split in understanding.

Reply:
from our standard, we should `useCallback ` for all functions. so the functions not using `useCallback ` should add a comment to explain why instead

Reply:
Let's take this offline, for now let's not block this pull request on this.